src/simple_pathfinding.cpp

```diff
@@ -291,9 +291,23 @@ simple_path<tripoint_abs_omt> find_overmap_path( const tripoint_abs_omt &source,
     open_set.push( scored_address{ start, 0 } );
     const point_abs_omt source_point = source.xy();
     constexpr int max_search_count = 100000;
+
+    auto last_report = std::chrono::high_resolution_clock::now();
+    int progress = 0;
     while( !open_set.empty() ) {
         const node_address cur_addr = open_set.top().addr;
         open_set.pop();
+        if( progress++ >= 100 ) { // stagger progress checks to 1 in 100 nodes
+            progress = 0;
+            if( progress_fn != nullptr ) {
+                const auto now = std::chrono::high_resolution_clock::now();
```

Comment:
According to https://en.cppreference.com/w/cpp/chrono/high_resolution_clock, `high_resolution_clock` is not always steady (i.e. time only goes forward and is unaffected by time sync). If we use `high_resolution_clock`, and the user's system happens to sync time backwards during pathfinding, they may again experience a freeze, so `steady_clock` should be used instead.

Reply:
It appears to be [aliased to steady_clock since VS 2015](https://learn.microsoft.com/en-us/cpp/standard-library/chrono?view=msvc-170#:~:text=high_resolution_clock%20is%20now%20a%20typedef%20for%20steady_clock%20In%20the%20Microsoft%20C%2B%2B%20implementation%2C), not sure about other compilers though, I changed it to steady_clock

Other parts of cata codebase uses a bunch of `high_resolution_clock`; specifically sdl sounds has a bunch of those which is where I peeked at trying to measure intervals, those worth fixing? They did seem to work fine for the most part